eth/blockwatch/block_watcher.go

```diff
@@ -281,28 +284,43 @@ func (w *Watcher) addLogs(header *MiniHeader) (*MiniHeader, error) {
 func (w *Watcher) getMissedEventsToBackfill(ctx context.Context) ([]*Event, error) {
 	events := []*Event{}
 
+	var (
+		startBlockNum          int
+		latestRetainedBlockNum int
+		blocksElapsed          int
+	)
+
 	latestRetainedBlock, err := w.stack.Peek()
 	if err != nil {
 		return events, err
 	}
-	// No blocks stored, nowhere to backfill to
-	if latestRetainedBlock == nil {
-		return events, nil
-	}
+
 	latestBlock, err := w.client.HeaderByNumber(nil)
 	if err != nil {
 		return events, err
 	}
-	blocksElapsed := big.NewInt(0).Sub(latestBlock.Number, latestRetainedBlock.Number)
-	if blocksElapsed.Int64() == 0 {
+	latestBlockNum := int(latestBlock.Number.Int64())
+
+	if latestRetainedBlock != nil {
```

Comment:
This piece of code will currently backfill starting from `lastRetainedBlock` as long as `lastRetainedBlock != nil`, but we actually want to do the following instead:

- If `lastRetainedBlock < currentRoundStartBlock`, backfill from `currentRoundStartBlock`
- Else, backfill from `lastRetainedBlock`

Given that this logic is reflected in how `backfillStartBlock` is initialized in `cmd/livepeer/livepeer.go`, in this piece of code, we could just check if `w.backfillStartBlock != nil` first - in effect, always backfilling from `backfillStartBlock` if it is passed in and if it is not passed in we would backfill from `lastRetainedBlock`

Reply:
Fixed in 95f4527f